client/pagination.go

```diff
@@ -0,0 +1,67 @@
+package client
+
+import "strconv"
+
+const limit = 100
+
+type Pagination struct {
+	offset int
+	params map[string]string
+}
+
+// Note: all the FIXME should be uncommented when Alpine adds support for go 1.18.
+// This will enable generics.
+
+type Paginated interface {
+	// FIXME: Environment
+	getEndpoint() string
+}
+
+func (p *Pagination) getParams() map[string]string {
+	return map[string]string{
+		"limit":  strconv.Itoa(limit),
```

Comment:
if no limit is passed in the params, what will happen? we didn't define default  a limit for the provider, didn't see one defined in this PR

Reply:
If no limit is passed it will use the default as defined by the API (in the website) (though currently missing from the website due to an issue). Regardless, it's better to be explicit and always pass a limit (in case the default changes in the backend). 

Sorry. I'm not sure I understand.
What do you mean to define a limit? where?

There's a `const limit = 100` in the code if that's what you meant...

Reply:
missed it, my bad

Reply:
@sagilaufer1992 - no worries. I just merged the code.
Regardless, if you think there are things that need to be improved (in general for environments), let me know and I will open a new issue.